src/Swashbuckle.AspNetCore.SwaggerGen/Generator/JsonPropertyExtensions.cs

```diff
@@ -13,14 +14,20 @@ internal static bool IsRequired(this JsonProperty jsonProperty)
         {
             return jsonProperty.Required == Newtonsoft.Json.Required.AllowNull
                 || jsonProperty.Required == Newtonsoft.Json.Required.Always
-                || jsonProperty.HasAttribute<RequiredAttribute>();
+                || jsonProperty.HasAttribute<RequiredAttribute>()
+                || jsonProperty.HasAttribute<BindRequiredAttribute>();
```

Comment:
As far as I know, `BindRequired` and `BindNever` are not applicable within types that are bound `FromBody`. Instead, serialization/deserialization for these is handled by input/output formatters. I'm not aware of an deserializer that's aware of MVC model binding attributes.

So, if the framework isn't going to honor these attributes when serializing/deserializing payloads, then I don't think SB should honor them when describing those payloads.